docker/build-project.sh

```
@@ -25,6 +25,12 @@ cd $WORKSPACE
 # get git commit ID of the project for using in buildhistory tagging
 CI_GIT_COMMIT=$(git rev-parse HEAD)
 
+# prepare git config
+#NOTE: bitbake does not pass git enviornment to tasks(do_patch),
+# hence we need to setup the git configuration 
+git config --global user.name "$GIT_COMMITTER_NAME"
+git config --global user.email "$GIT_COMMITTER_EMAIL"
+
 # FIXME: undbound variables used without checking:
```

Don't do that. Patches must have origin information. This will hide that error situation.

---

@Kad, This is not just related to git apply or patch formatting, As i mentioned in commit message the real issue is with the git environment in bitbake tasks, though we set GIT_COMITTER_{NAME/EMAIL},  those are not available to while patching source. So here the idea is to pass them via 'git config'

---

If patch in your changeset would have proper information in header, those wouldn't be needed.

---

@kad, This issue raised on yocto project by @mythi and they accepted to fix in oe-core. Hopefully we need not to set this configuration as oecore itself provide dummy git config.

We can revert this once the fix is available.

---

let's get fix from oe-core (Paul's change to patch functions). it's better than trying to create local workarounds.

---

Paul's fix your are referring(http://git.yoctoproject.org/cgit/cgit.cgi/poky/commit/?id=4b4387455c62cf19fa3d215a11b5d0b1211d4570) does not fix kernel patching issue, that address only PATCHTOOL='git' case.

This is the fix we needed:
http://lists.openembedded.org/pipermail/openembedded-core/2016-October/127230.html

---

in reality, I think both of those patches are needed. we stepped on one case and issue present in different scenarios.